src/cladetime/cladetime.py

```diff
@@ -210,3 +212,102 @@ def _get_config(self) -> Config:
+        if sequence_count == 0:
+            msg = "sequence_metadata is empty \n" "Stopping clade assignment...."
+            warnings.warn(
+                msg,
+                category=CladeTimeSequenceWarning,
+            )
+            return pl.LazyFrame()
+
+        # if there are many sequences in the filtered metadata, warn that clade assignment will
+        # take a long time and require a lot of resources
+        if sequence_count > self._config.clade_assignment_warning_threshold:
+            msg = (
+                f"Sequence count is {sequence_count}: clade assignment will run longer than usual. "
+                "You may want to run clade assignments on smaller subsets of sequences."
+            )
+            warnings.warn(
+                msg,
+                category=CladeTimeSequenceWarning,
+            )
+
+        logger.info(
+            "Starting clade assignment pipeline",
+            sequence_as_of=self.sequence_as_of,
+            tree_as_of=self.tree_as_of,
+            num_sequence=sequence_count,
+        )
+
+        # drop any clade-related columns from sequence_metadata (if any exists, it will be replaced
+        # by the results of the clade assignment)
+        sequence_metadata = sequence_metadata.drop(
+            [
+                col
+                for col in sequence_metadata.collect_schema().names()
+                if col not in self._config.nextstrain_standard_metadata_fields
+            ]
+        )
+
+        ids = sequence.get_metadata_ids(sequence_metadata)
+        tree = Tree(self.tree_as_of, self.url_sequence)
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            filtered_sequences = sequence.filter(ids, self.url_sequence, Path(tmpdir))
+            nextclade_dataset = _get_nextclade_dataset(
+                tree.ncov_metadata.get("nextclade_version_num"),
+                tree.ncov_metadata.get("nextclade_dataset_name").lower(),
+                tree.ncov_metadata.get("nextclade_dataset_version"),
+                Path(tmpdir),
+            )
+            logger.info(
+                "Assigning clades",
+                sequences_to_assign=len(ids),
+                nextclade_dataset_version=tree.ncov_metadata.get("nextclade_dataset_version"),
+            )
+            assignments = _get_clade_assignments(
+                tree.ncov_metadata.get("nextclade_version_num"), filtered_sequences, nextclade_dataset, output_file
+            )
+            logger.info(
+                "Clade assignments done",
+                assignment_file=assignments,
+                nextclade_dataset=tree.ncov_metadata.get("nextclade_dataset_version"),
+            )
+
+            assigned_clades = pl.read_csv(assignments, separator=";", infer_schema_length=100000)
+
+        assigned_clades = sequence_metadata.join(
+            assigned_clades.lazy(), left_on="strain", right_on="seqName", how="left"
+        )
+        return assigned_clades
```

Comment:
Future commits change the return value to an object that includes summarized clade counts as well as the line file